common/flatpak-dir.c

```diff
@@ -6946,14 +6946,57 @@ flatpak_dir_undeploy_all (FlatpakDir   *self,
   return TRUE;
 }
 
+/**
+ * flatpak_dir_remove_ref:
+ *
+ * Remove the flatpak ref given by @remote_name:@ref from the underlying
+ * OSTree repo. Attempting to remove a ref that is currently deployed
+ * is an error, you need to uninstall the flatpak first. Note that this does
+ * not remove the objects bound to @ref from the disk, you will need to
+ * call flatpak_dir_prune to do that.
+ *
+ * @self: a #FlatpakDir
+ * @remote_name: the name of the remote
+ * @ref: the flatpak ref to remove
+ * @cancellable: (allow-none): a #GCancellable
+ * @error: (allow-none): a #GError
```

Comment:
Drop the `(allow-none)` annotation from `@error`. It’s not needed (`g-ir-scanner` already knows about the conventions for `GError` arguments) and is deprecated. In this case, in particular, `(allow-none)` would be replaced by `(nullable) (optional)`, thus demonstrating the difference between the two replacements for `(allow-none)`. `(nullable)` on an out argument means `NULL` may be passed out; `(optional)` on an out argument means `NULL` may be passed in.